astropy/io/fits/fitsrec.py

```diff
@@ -628,7 +628,7 @@ def _coldefs(self):
         try:
             del self.__dict__['_coldefs']
         except KeyError as exc:
-            raise AttributeError(exc.args[0])
+            pass
```

Comment:
This isn't what I suggested--I wrote that I still want `AttributeError` to be raised here.  The possible `AttributeError` should just be caught in the `__del__` method below.  Basically `@_coldefs.deleter` doesn't need to be changed at all.  I gave the exact prescription here: https://github.com/astropy/astropy/pull/4694#discussion_r56163061

Thanks for working on this--this is a bad bug and I'm glad you caught it. Even if I'm being nitpicky your initial fix was not wrong and helped to quickly illustrate where the problem was :)